Glasgow | ITP May 2025 | Adiyah Farhan | Sprint 3 | Alarm Clock

AdiyahFarhan opened the pull request:
… acceptance criteria

<!--

You must title your PR like this:

Region | Cohort | FirstName LastName | Sprint | Assignment Title

For example,

London | 25-ITP-May | Carol Owen | Sprint 1 | Alarm Clock

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [X] I have committed my files one by one, on purpose, and for a reason
- [X] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [X] I have tested my changes
- [X] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [X] My changes meet the [requirements](./README.md) of this task

## Changelist

Implemented the setAlarm() function in alarm clock according to given acceptance criteria

## Questions

No

arun-john commented:
Hello Adiyah, good effort on this and nice to see the logic broken out into functions with good comments!

Did you try running the automated tests with `npm test`? I can see some failures here
I see you've tried to implement the color change / flashing logic from Extra Tasks. Color change seems to be working well, but probably needs a reset when starting the alarm again. Flashing logic needs more changes, in case you were planning to attempt that...

We are also missing some `index.html` changes (Pls. refer back to `readme.md` for details)

Good luck with the changes!

AdiyahFarhan commented:
Thank you, @arun-john, for taking the time to review my pull request. I have reviewed your feedback and updated the files accordingly. The test failure was due to a  typo error, which I have now corrected. Please check. Thanks

arun-john commented:
Great, seems all good!

One final thing - Try testing this out manually by setting the time to say 30 (and click Set Alarm). While the clock is counting down, set it to a different value, say 5 and click Set Alarm. Check what happens.

arun-john commented:
All working...good work on this Adiyah! :+1:
